### PR TITLE
RFC: Docs draft

### PR DESCRIPTION
**Pull request by vuil**
### What this PR does / why we need it

Draft of the repo documentation for review.
The larger set of changes are in https://github.com/vmware-tanzu/tanzu-cli/pull/137, of which the plugin development guide will make references back to the docs here.

### Which issue(s) this PR fixes

Fixes #N/A

### Release note
```release-note
NONE
```

### CODE REVIEW

**Review by mpanchajanya — APPROVED** [review]
LGTM!. I will update the Config APIs

File: README.md [hunk 1]
```diff
@@ -8,23 +8,67 @@ watch this page for further updates.
...
+the development process, see the (VVV update link) [Tanzu CLI Plugin Development guide](https://github.com/vuil/tanzu-cli/blob/docs-draft/docs/dev/main.md)
 
-Tanzu Plugin Runtime provides various config API methods to perform CRUD
-operations on Contexts, Servers, DiscoverySources, Features, Envs etc.
+## The library
+
+This Tanzu Plugin Runtime broadly consists of:
+
+1. CLI UX Component library
+1. Config library
```
Comment (marckhouzam) [inline README.md:26]: Config -> Configuration?

File: README.md [hunk 2]
```diff
@@ -8,23 +8,67 @@ watch this page for further updates.
...
+- output writers (table, listtable, json, yaml, spinner)
+- prompt
+- selector
+- question
+
+### Config Library
+
+This package implements helper functions to read, write and update various
+Tanzu CLI configuration objects like Context's, DiscoverySource's, CLI
+Feature's and environment settings.
```
Comment (marckhouzam) [inline README.md:44]: Why the apostrophe-s for each item?

File: README.md [hunk 3]
```diff
@@ -8,23 +8,67 @@ watch this page for further updates.
...
+This package implements helper functions to develop test plugins.
+
+VVV provide links to more details for these packages.
+
+## Testing
+
+Besides unit and integration tests, the runtime APIs are also being tested
+in the presence of other versions runtime client code. These tests are
+important in ensuring interoperability among runtime clients (and hence
+different generations of CLI plugins)
```
[the remark this answers is not in the document]
Comment (marckhouzam) [reply README.md:69]: So this morning, I don't understand that sentence anymore 🤯 
Are we really testing interoperability between runtimes?  Or are we testing interoperability between different CLI versions and different runtime?  

I realize that the CLI uses the runtime, which may explain the phrasing, but I don't think this is something users will necessarily grasp.  I think the interop between different CLI versions and plugins using different runtime versions is more understandable.

File: README.md [hunk 4]
```diff
@@ -8,23 +8,67 @@ watch this page for further updates.
...
 
-### Overview
+### Cross Version API Compatibility Testing
```
Comment (marckhouzam) [inline README.md:71]: Is this general API compatibility testing, or just for the "configuration" handling.  The first paragraph seems to say it is only for configuration, but the rest of the section and title seems broader.

File: docs/config.md [hunk 5]
```diff
@@ -20,6 +20,8 @@ Note: due to the fact that information in the CFG_NG file directly affects the m
 
 ### Runtime Config APIs
 
+VVV Update list, remove references to 'Server'
```
[the remark this answers is not in the document]
Comment (marckhouzam) [reply docs/config.md:23]: cc @mpanchajanya

**Review by marckhouzam — COMMENTED** [review]
Some suggested small tweaks

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mpanchajanya) [reply README.md:69]: This is referring to Cross-version API compatibility testing where we test whether different versions of Runtime APIs can inter operate

File: docs/config.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (mpanchajanya) [reply docs/config.md:23]: @marckhouzam Details section at the top of this doc has intro on these terms https://github.com/vmware-tanzu/tanzu-plugin-runtime/blob/f3730c5a33f7e4fac60e10be441972534c0ae535/docs/config.md#details .

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (vuil) [reply README.md:69]: But point taken if there is a way to phrase the section more clearly, don't hesitate to propose something.
Interop between CLI versions are also being tested in the tanzu-cli repo.

[the remark this answers is not in the document]
Comment (vuil) [reply README.md:69]: @mpanchajanya to provide more details and link to existing docs about these tests.

File: docs/config.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (vuil) [reply docs/config.md:23]: There is something worth improving upon, cause the top of the text reads a little weird after much of the design details behind these files were moved to docs/design/config_ng.md. 

The glossary at the top can still be there, but it feels a little abrupt. I suggest we furnish some intro about the CLI configuration just before it.

File: README.md (hunk 4 above)
Comment (vuil) [reply README.md:71]: @mpanchajanya to followup

File: README.md [hunk 6]
```diff
@@ -8,23 +8,67 @@ watch this page for further updates.
...
+### Test Helpers
+
+This package implements helper functions to develop test plugins.
+
+VVV provide links to more details for these packages.
+
+## Testing
+
+Besides unit and integration tests, the runtime APIs are also being tested
+in the presence of other versions runtime client code. These tests are
```
[the remark this answers is not in the document]
Comment (vuil) [reply README.md:67]: We are trying to convey this tests should help validate interooperatibility of code utilizing runtime (e.g. plugin code, hence runtime clients). Maybe there is a better to phrase this, but I would keep for now and address in followup if necessary.

**Comment by mpanchajanya** [comment]
@vuil @chandrareddyp @marckhouzam  I will address the comments in my [PR-40](https://github.com/vmware-tanzu/tanzu-plugin-runtime/pull/40)

**Pull request merged**